### PR TITLE
[OREE-1874] add stage to opportunity contextual params

### PR DESCRIPTION
**Pull request by sheff3rd**
[OREE-1874](https://outreach-io.atlassian.net/browse/OREE-1874)

-------

Add support for additional opportunity parameter - stage

[OREE-1784]: https://outreach-io.atlassian.net/browse/OREE-1784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OREE-1874]: https://outreach-io.atlassian.net/browse/OREE-1874?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/OpportunityContextUtils.test.ts [hunk 1]
```diff
@@ -223,4 +223,174 @@ describe('OpportunityContextUtils', () => {
...
   });
+
+  describe('external provider properties', () => {
+    describe('initOpportunityContext', () => {
+      it('external created at context will be set', () => {
+        initOpportunityContext(ctx, {
+          key: OpportunityContextKeys.EXTERNAL_CREATED_AT,
+          value: new Date('2021-01-01'),
+        });
+        expect(ctx.externalCreatedAt).toBe('some-value');
```
Comment (palootcenas-outreach) [inline tests/OpportunityContextUtils.test.ts:234]: You will need to use `jest.useFakeTimers('modern')` to test dates confidently.

Also `new Date('2021-01-01').toString() !== 'some-value'` so this test will fail.

**Comment by sheff3rd** [comment]
@palootcenas-outreach 
I'm sorry but I were not able to utilize `jest.useFakeTimers('modern')` in any way to run tests
no matter what I tried, there was discrepancy between CI and local timezones

I've tried tried this for an hour, but not to waste any more time I opted for a different solution

**Comment by palootcenas-outreach** [comment]
@sheff3rd It's fine if it works in CI.

File: tests/OpportunityContextUtils.test.ts [hunk 2]
```diff
@@ -223,4 +223,176 @@ describe('OpportunityContextUtils', () => {
...
+      it('external provider name context will be set', () => {
+        initOpportunityContext(ctx, {
+          key: OpportunityContextKeys.EXTERNAL_PROVIDER,
+          value: 'some-value',
+        });
+        expect(ctx.externalProviderName).toBe('some-value');
+      });
+    });
+
+    describe('toOpportunityParams', () => {
```
Comment (honeydripper) [inline tests/OpportunityContextUtils.test.ts:252]: From here and below it feels way to repetitive, worth refactoring at some point in the future.

**Review by honeydripper — APPROVED** [review]

**Pull request merged**